rust/pattern/constraint/type_constraint.rs

```diff
@@ -223,10 +216,10 @@ impl From<bool> for IsKey {
 
 pub const KEY: IsKey = IsKey::Yes;
 
-impl Display for IsKey {
-    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+impl fmt::Display for IsKey {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
         if *self == IsKey::Yes {
-            f.write_str(" @key")?;
+            write!(f, " {}", IsKey)?;
```

Comment:
May be a little confusing: this is `token::Constraint::IsKey`, which shares its name with `enum IsKey` from this file. It's unambiguous from the compiler's point of view.